lib/OpenQA/Task/Utils.pm

```diff
@@ -48,5 +51,30 @@ sub finish_job_if_disk_usage_below_percentage (%args) {
     return 1;
 }
 
+# define a signal handler for retrying the job after receiving a signal
+my $CONCLUDE_ON_SIGNALS = 0;
+sub _handle_signal ($job, $signal) {
+    # do nothing if the job is supposed to be concluded despite receiving a signal at this point
+    return $job->note(signal_handler => "Received signal $signal, concluding") if $CONCLUDE_ON_SIGNALS;
+
+    # schedule a retry before stopping the job's execution prematurely
+    $job->note(signal_handler => "Received signal $signal, scheduling retry and releasing locks");
+    $job->retry;
+    exit;
+}
+
+# enables retrying the specified Minion job when receiving SIGTERM/SIGINT
+# note: Prevents the job to fail with "Job terminated unexpectedly".
+sub enable_retry_on_signals ($job) {
+    my $handler = sub ($signal) { _handle_signal($job, $signal) };
+    $SIG{TERM} = $handler;
+    $SIG{INT} = $handler;
```

Comment:
I guess this could be

```suggestion
    $SIG{INT} = $SIG{TERM} = sub ($signal) { _handle_signal($job, $signal) };
```

Reply:
I can try but it might not work because `SIG` is a special variable.

Reply:
Should work.